torch/ao/quantization/fx/prepare.py

```diff
@@ -514,15 +580,16 @@ def _get_arg_target_is_dynamic_as_input_to_node(
     to node `node`
     """
     assert isinstance(arg, Node)
-    is_weight = node_arg_is_weight(node, arg, backend_config)
-    is_bias = node_arg_is_bias(node, arg, backend_config)
-    is_activation = not is_weight and not is_bias
-    if is_activation and "input_act_obs_or_fq_ctr" in node.meta["target_dtype_info"]:
-        input_act_obs_or_fq_ctr = node.meta["target_dtype_info"].get("input_act_obs_or_fq_ctr")
-        _, qconfig_is_dynamic = _get_dtype_and_is_dynamic(input_act_obs_or_fq_ctr)
-        return qconfig_is_dynamic
-    else:
-        return False
+    assert "target_dtype_info" in node.meta
+    assert "input_obs_or_fq_ctr_for_args" in node.meta["target_dtype_info"]
+    input_obs_or_fq_ctr_for_args = node.meta["target_dtype_info"]["input_obs_or_fq_ctr_for_args"]
+    arg_to_ctr = {}
+    _populate_arg_to_ctr_for_args(node.args, input_obs_or_fq_ctr_for_args, arg_to_ctr)
+    if "_input_obs_or_fq_ctr_for_kwargs" in node.meta["target_dtype_info"]:
+        _populate_arg_to_ctr_for_kwargs(node.kwargs, node.meta["target_dtype_info"]["_input_obs_or_fq_ctr_for_kwargs"], arg_to_ctr)
```

Comment:
As before, we are repeatedly populating the dictionary. If you think this will be used later then consider refactoring in subsequent diff.

Reply:
sounds good, I'll refactor this